library/src/network.rs

```diff
@@ -185,6 +188,7 @@ impl PatchCheckRequest {
             release_version: config.release_version.clone(),
             platform: current_platform().to_string(),
             arch: current_arch().to_string(),
+            rollout_group: UpdaterState::load_or_new_from_config(config).rollout_group(),
```

Comment:
Should this be passed a config?  Do we need to hold the config lock to call this?

Reply:
Our convention throughout the codebase has been that you get a config with `with_config`, which obtains the lock, and that anything that receives a config as a parameter assumes that the config it receives is properly locked.